train.py

```diff
@@ -264,11 +277,14 @@ def distributed_main(i, args, start_rank=0):
         args.distributed_rank = start_rank + i
     main(args, init_distributed=True)
 
+def parse_input_shapes(input_shapes_arg):
+    input_shapes = (
+        shape.replace('*', 'x').split('x') for shape in input_shapes_arg)
+    input_shapes = [list(map(int, shape)) for shape in input_shapes]
+    input_shapes.sort(key=lambda shape: shape[1])
+    return input_shapes
 
-def cli_main():
-    parser = options.get_training_parser()
-    args = options.parse_args_and_arch(parser)
-
+def cli_main_gpu(args):
```

Comment:
same comments as above. can we just preserve `cli_main` and fold `tpu` options in here (and if needed remove `gpu` options